addons/payment/models/payment_acquirer.py

```diff
@@ -270,952 +309,166 @@ def _prepare_account_journal_vals(self):
             # Show the journal on dashboard if the acquirer is published on the website.
             'show_on_dashboard': self.state == 'enabled',
-            # Don't show payment methods in the backend.
+            # Don't show payment methods in the backend
             'inbound_payment_method_ids': inbound_payment_method_ids,
             'outbound_payment_method_ids': [],
         }
 
+    #=== BUSINESS METHODS ===#
+
     @api.model
-    def _create_missing_journal_for_acquirers(self, company=None):
-        '''Create the journal for active acquirers.
-        We want one journal per acquirer. However, we can't create them during the 'create' of the payment.acquirer
-        because every acquirers are defined on the 'payment' module but is active only when installing their own module
-        (e.g. payment_paypal for Paypal). We can't do that in such modules because we have no guarantee the chart template
-        is already installed.
-        '''
-        # Search for installed acquirers modules.
-        # If this method is triggered by a post_init_hook, the module is 'to install'.
-        # If the trigger comes from the chart template wizard, the modules are already installed.
-        acquirer_modules = self.env['ir.module.module'].search(
-            [('name', 'like', 'payment_%'), ('state', 'in', ('to install', 'installed'))])
-        acquirer_names = [a.name.split('_', 1)[1] for a in acquirer_modules]
+    def _get_compatible_acquirers(
+        self, company_id, partner_id, allow_tokenization=False, preferred_acquirer_id=None,
+        **_kwargs
+    ):
+        """ Select and return the acquirers matching the criteria.
+
+        The base criteria are that acquirers must not be disabled, be in the company that is
+        provided, and support the country of the partner if it exists.
+
+        If a `preferred_acquirer_id` is specified, only the corresponding acquirer is returned.
+        If that acquirer does not exist or does not match the criteria, only the acquirers that
+        do match them are returned.
+
+        :param int company_id: The company to which acquirers must belong, as a `res.company` id
+        :param int partner_id: The partner making the payment, as a `res.partner` id
+        :param bool allow_tokenization: Whether matching acquirers must allow tokenization
+        :param int preferred_acquirer_id: The preferred acquirer, as a `payment.acquirer` id
+        :param dict _kwargs: Optional data. This parameter is not used here
+        :return: The compatible acquirers
+        :rtype: recordset of `payment.acquirer`
+        """
+        # Compute the base domain for compatible acquirers
+        domain = ['&', ('state', 'in', ['enabled', 'test']), ('company_id', '=', company_id)]
+        partner = self.env['res.partner'].browse(partner_id)
+
+        # Handle partner country
+        if partner.country_id:  # The partner country must either not be set or be supported
```

Comment:
In that case we don't filter out the acquirer since we're unsure of the partner's country.
This is open for debate but this is aligned with the general philosophy of the payment module that is "don't prevent a customer to pay, unless the acquirer is specifically configured against it".